animation_nodes/nodes/text/decompose_text.py

```diff
@@ -0,0 +1,107 @@
+import bpy
+import blf
+from bpy.props import *
+from mathutils import Matrix
+from string import whitespace
+from functools import lru_cache
+from ... events import propertyChanged
+from ... data_structures import Matrix4x4List
+from ... base_types import AnimationNode, VectorizedSocket
+
+class DecomposeTextNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_DecomposeTextNode"
+    bl_label = "Decompose Text"
+    errorHandlingType = "EXCEPTION"
+
+    includeWhiteSpaces : BoolProperty(name = "Include White Spaces", default = True,
+        update = propertyChanged)
+
+    def create(self):
+        self.newInput("Text", "Text", "text")
+        self.newInput("Font", "Font", "font")
+        self.newInput("Float", "Character Spacing", "charSpacing", value = 1, hide = True)
+        self.newInput("Float", "Word Spacing", "wordSpacing", value = 1, hide = True)
+        self.newInput("Float", "Line Spacing", "lineSpacing", value = 1, hide = True)
+        self.newInput("Text", "Alignment", "align", value = "LEFT", hide = True)
+
+        self.newOutput("Matrix List", "Transforms", "transforms")
+        self.newOutput("Text List", "Characters", "characters")
+        self.newOutput("Integer", "Length", "length")
+
+    def drawAdvanced(self, layout):
+        layout.prop(self, "includeWhiteSpaces")
+
+    def execute(self, text, font, charSpacing, wordSpacing, lineSpacing, align):
+        self.validateAlignment(align)
+        fontID = self.getFontID(font)
+        fontRatio = self.getFontRatio(font, fontID)
+
+        return self.getCharTransforms(text, fontID, fontRatio, charSpacing, wordSpacing, lineSpacing, align)
+
+    def getCharTransforms(self, text, fontID, fontRatio, charSpacing, wordSpacing, lineSpacing, align):
+        allChars = []
+        allTransforms = Matrix4x4List()
+        for i, line in enumerate(text.splitlines()):
+            chars = []
+            transforms = []
+            comulativeWidth = 0
+            for char in line:
+                if self.includeWhiteSpaces or char not in whitespace:
+                    transforms.append(Matrix.Translation((comulativeWidth, -i * lineSpacing, 0)))
+                    chars.append(char)
+
+                comulativeWidth += self.getCharWidth(fontID, char, fontRatio, charSpacing, wordSpacing)
+
+            transforms = Matrix4x4List.fromValues(transforms)
+
+            if align == "CENTER":
+                offset = (-comulativeWidth + charSpacing / 2 - 0.5) / 2
+                transforms.transform(Matrix.Translation((offset, 0, 0)))
+            elif align == "RIGHT":
+                offset = -comulativeWidth + charSpacing / 2 - 0.5
+                transforms.transform(Matrix.Translation((offset, 0, 0)))
+
+            allChars += chars
+            allTransforms.extend(transforms)
+
+        return allTransforms, allChars, len(allChars)
+
+    def getCharWidth(self, fontID, char, fontRatio, charSpacing, wordSpacing):
```

Comment:
`getCharWidth` and `getFontRatio` can be pulled out of the class, because they do not depend on `self`. The `lru_cache` might cache the same font multiple times currently, because of the `self` parameter.